Automattic/newspack-ios
# Issues/48 current working folder

Closes #48 

This PR changes the way the current working / selected story folder is identified.  Changes include:
- Functionality that was in the AssetStore is moved to the FolderStore and the AssetStore (and friends) are removed. 
- There will always be at least one story folder. If the last story folder is deleted a new one is created and made the current story folder.
- Deleting the current story folder will automatically select a different story folder as the current one.
- For convenience, a different disclosure icon identifies the current story folder.

To test: 
- Run all tests and ensure they pass. 

Scenario 1
- Launch the app and view the folder list.
- Create some folders if you do not currently have any.
- Observe one of the folders is indicated as the current story by a different disclosure icon.

Scenario 2
- Delete the selected story.  Ensure another is automatically chosen. 

Scenario 3
- Delete all folders.  Ensure that a new folder is created to replace the last one deleted and it is marked the selected folder.

Scenario 4
- Add several more folders. 
- Try selecting a folder, tapping back to the menu, then back to the folder list. Ensure the correct folder is still selected.  

Finally
- Check the files app and ensure the folders match what you end up having in the app.

@jleandroperez  Can I trouble you with what is probably the final PR for this milestone?
